framework/src/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesHelper.java

```diff
@@ -221,15 +221,36 @@ private void removeIgnoredAnnosFromScene(AScene scene) {
         for (AClass aclass : scene.classes.values()) {
             for (AField field : aclass.fields.values()) {
                 removeIgnoredAnnosFromATypeElement(field.type, TypeUseLocation.FIELD);
+                if (field.type.innerTypes.size() != 0) {
```

Comment:
Since this identical code appears 4 times, how about ceating a `removeIgnoredAnnosFromATypeElementRecursively` routine that consists of the call to `removeIgnoredAnnosFromATypeElement` and the following 5 lines?